e2etest/utils.go

```diff
@@ -25,3 +30,26 @@ func runCommand(bgp bool, name string, args ...string) (string, error) {
 	out, err := exec.Command(name, args...).CombinedOutput()
 	return string(out), err
 }
+
+func wgetRetry(bgp bool, address string) error {
+	retrycnt := 0
+	code := 0
+	var err error
+
+	// Retry loop to handle wget NetworkFailure errors
+	for {
+		_, err = runCommand(bgp, "wget", "-O-", "-q", address, "-T", "60")
+		if exitErr, ok := err.(*exec.ExitError); err != nil && ok {
```

Comment:
There's a check `err != nil`, but wouldn't this break if `err == nil` because of `err.(*exec.ExitError)` ?

but the tests are passing ... what am I missing :)

Reply:
if there is no err "nil" we will break from for loop ? maybe I don't understand the question ? basically if we have err and we are able to extract the exit code then read the code and check for NetworkFailure

Reply:
nevermind, i got it sorted out